### PR TITLE
Subscriptions: Action delegation

### PR DESCRIPTION
**Pull request by prateekbh**
- Add action delegation via attribute

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 1]
```diff
@@ -437,17 +437,26 @@ export class SubscriptionService {
    * @return {!Promise<boolean>}
    */
   delegateActionToLocal(action) {
-    const localPlatform = /** @type {LocalSubscriptionPlatform} */ (
-      dev().assert(this.platformStore_.getLocalPlatform(),
-          'Local platform is not registered'));
-    // TODO: add which service is passing this event
+    this.delegateActionToService(action, 'local');
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: `return` missing? It's declared as `@return {!Promise...}`. That, I'd expect, should be "noticed" by a test?

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 2]
```diff
@@ -437,17 +437,26 @@ export class SubscriptionService {
...
+  }
+
+  /**
+   * Delegates an action to specified platform
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: nit: end sentence with "."

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 3]
```diff
@@ -437,17 +437,26 @@ export class SubscriptionService {
...
+   * @param {string} action
+   * @param {string} serviceId
+   * @return {!Promise<boolean>}
+   */
+  delegateActionToService(action, serviceId) {
+    const platform = dev().assert(this.platformStore_.getPlatform(serviceId),
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: This was ok as a local that's available sync. But will break regularly with other platforms loaded in separate extensions. I.e. `getPlatform('subscribe.google.com')` will fail if the extension hasn't loaded yet. We'd need to use a promisable API for this. It's ok to start with a `TODO` here and revisit in a follow up PR.

File: extensions/amp-subscriptions/0.1/service-adapter.js [hunk 4]
```diff
@@ -39,7 +39,17 @@ export class ServiceAdapter {
    * @return {!Promise<boolean>}
    */
   delegateActionToLocal(action) {
-    return this.subscriptionService_.delegateActionToLocal(action);
+    return this.delegateActionToService(action, 'local');
+  }
+
+  /**
+   * Delegates actions to local platform
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/service-adapter.js]: nit: "."

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Done

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Done

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Done

File: extensions/amp-subscriptions/0.1/service-adapter.js (hunk 4 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/service-adapter.js]: Done

**Pull request merged**